### PR TITLE
131538: Page to Delete Nti

### PR DESCRIPTION
**Pull request by blakebyron-nimble**
**What is the change?**
Add ability for the User to Delete and NTI

**Why do we need the change?**
Allow users to delete incorrectly added data

**What is the impact?**
Users can tidy up information, information will not be completely removed. Page will be released behind feature flag 

**Azure DevOps Ticket**
[131538](https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/131538)

### CODE REVIEW

**Comment by blakebyron-nimble** [comment]
Note, there are no cypress tests in this PR as I don't think we'll. Please let me know if we'd prefer them as part of this PR/separately or at all.

File: ConcernsCaseWork/ConcernsCaseWork/Services/Nti/NtiModelService.cs [hunk 1]
```diff
@@ -92,5 +92,10 @@ public async Task StoreNtiAsync(NtiModel ntiModel, string continuationId)
 
 			await _ntiCachedService.SaveNtiAsync(NtiMappers.ToDBModel(ntiModel), continuationId);
 		}
+
+		public async Task DeleteAsync(long ntiId)
+		{
+			await _ntiCachedService.DeleteNtiAsunc(ntiId);
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Services/Nti/NtiModelService.cs:98]: Typo

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Nti/Delete.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+				}
+				if (model != null && model.CaseUrn != this.CaseUrn)
+				{
+					return NotFound();
+				}
+			}
+			catch (HttpRequestException ex)
+			{
+				_logger.LogErrorMsg(ex);
+				if (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Nti/Delete.cshtml.cs:54]: QUESTION:

Do we need to handle this separately? Does it need to show a different page, if its only us that will see it?

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Nti/Delete.cshtml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+@using NetEscapades.AspNetCore.SecurityHeaders
+@model ConcernsCaseWork.Pages.Case.Management.Action.Nti.DeletePageModel
+
+@{
+    ViewData["Title"] = "Delete NTI";
+    var nonce = HttpContext.GetNonce();
+    ViewData[ViewDataConstants.CancelBackLink] = $"/case/{@Model.CaseUrn}/management/action/nti/{Model.NtiId}";
+}
+
+<div class="govuk-width-container">
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/Nti/Delete.cshtml:13]: Do we need UI automation to prove this works?

**Review by mikestock-nimble — APPROVED** [review]

**Pull request closed without merging**